src/network/network_client.cpp

```diff
@@ -1188,6 +1188,8 @@ NetworkRecvStatus ClientNetworkGameSocketHandler::Receive_SERVER_CONFIG_UPDATE(P
 
 	_network_server_max_companies = p.Recv_uint8();
 	_network_server_name = p.Recv_string(NETWORK_NAME_LENGTH);
+
+	InvalidateWindowData(WC_CLIENT_LIST, 0);
```

Comment:
`InvalidateWindowData` also marks the window dirty, so the `SetWindowClassesDirty` can be removed.

Reply:
Thanks, I've updated it